universal-application-tool-0.0.1/app/services/cloud/azure/BlobStorageUploadRequest.java

```diff
@@ -0,0 +1,76 @@
+package services.cloud.azure;
+
+import com.google.auto.value.AutoValue;
+import services.cloud.StorageServiceName;
+import services.cloud.StorageUploadRequest;
+
+/** This holds the information needed to upload a file to blob storage. */
+@AutoValue
+public abstract class BlobStorageUploadRequest implements StorageUploadRequest {
+
+  public abstract String accountName();
+
+  public abstract String containerName();
+
+  public abstract String fileName();
+
+  public abstract String sasToken();
+
+  public abstract String blobUrl();
+
+  public abstract String successActionRedirect();
+
+  @Override
+  public abstract String serviceName();
+
+  public static Builder builder() {
+    return new AutoValue_BlobStorageUploadRequest.Builder()
+        .setServiceName(StorageServiceName.AZURE_BLOB.getString());
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Get file name for the blob. This is used to build the canonicalized resource part of the
+     * signature string.
+     */
```

Comment:
suggestion (blocking): move all doc comments into the outer main class rather than having them in the builder

Reply:
Done.